meshtastic/interfaces/ble/notifications.py

```
@@ -228,3 +283,413 @@ def _get_callback(self, characteristic: str) -> Callable[[Any, Any], None] | Non
+        def _tracked_handler_thunk() -> None:
+            nonlocal executed
+            executed = True
+            handler_thunk()
+
+        def _fallback_if_not_executed() -> None:
+            if not executed:
+                fallback()
+
+        try:
+            safe_execute(_tracked_handler_thunk, error_msg=error_msg)
+            return
```

Ruff (0.15.6)

[warning] Consider moving this statement to an `else` block

(TRY300)